Gem/Code/Source/Weapons/BaseWeapon.h

```diff
@@ -78,13 +83,22 @@ namespace MultiplayerSample
         const WeaponParams m_weaponParams;
 
         WeaponListener& m_weaponListener;
-        ClientEffect m_activateEffect;
-        ClientEffect m_impactEffect;
-        ClientEffect m_damageEffect;
-        FireParams   m_fireParams;
-        NetEntityIdSet m_gatheredNetEntityIds;
 
-        int32_t m_ammoSurfaceTypeIndex = -1;
+#if AZ_TRAIT_CLIENT
```

Comment:
Nit: consider adding comments to guide folks through the setup or whats been exposed.